File: src/main/java/net/hectus/neobb/turn/legacy_game/block/LTSeaPickle.java
package net.hectus.neobb.turn.legacy_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.NatureClazz;
import net.hectus.neobb.turn.default_game.attributes.function.AttackFunction;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import org.bukkit.block.Block;

import java.util.List;

public class LTSeaPickle extends BlockTurn implements AttackFunction, BuffFunction, NatureClazz {
    public LTSeaPickle(NeoPlayer player) { super(player); }
    public LTSeaPickle(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 0;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn(Buff.BuffTarget.NEXT, 2));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/legacy_game/block/LTSeaPickle.kt
package net.hectus.neobb.turn.legacy_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.NatureClazz
import net.hectus.neobb.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.turn.default_game.block.BlockTurn
import org.bukkit.block.Block

class LTSeaPickle(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackFunction, BuffFunction, NatureClazz {
    override fun buffs(): List<Buff<*>> {
        return listOf(ExtraTurn(2, Buff.BuffTarget.NEXT))
    }
}